aep/general/0162/aep.md.j2

```diff
@@ -1,5 +1,388 @@
+useful include:
+
+- When it is valuable to expose older revisions of a resource via an API. This
+  can avoid the overhead of the customers having to write their own API to store
+  and enable retrieval of revisions.
+- Other resources depend on or descend from different revisions of a resource.
+- There is a need to represent the change of a resource over time.
+
+APIs implementing resources with revisions **must** model resource
+revisions as a subresource of the resource.
+
+{% tab proto %}
+
+```proto
+message BookRevision {
+  // The path of the book revision.
+  string path = 1;
+
+  // The snapshot of the book
+  Book resource = 2
+    [(google.api.field_behavior) = OUTPUT_ONLY];
+
+  // The timestamp that the revision was created.
+  google.protobuf.Timestamp create_time = 3
+    [(google.api.field_behavior) = OUTPUT_ONLY];
+
+  // Other revision IDs that share the same snapshot.
+  repeated string aliases = 4
+    [(google.api.field_behavior) = OUTPUT_ONLY];
+}
+```
+
+- The `message` **must** be annotated as a resource (AIP-123).
+- The `message` name **must** be named `{ResourceType}Revision`.
+
+{% tab oas %}
+
+```json
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "type": "object",
+    "properties": {
+        "path": {
+            "type": "string"
+        },
+        "resource": {
+            "$ref": "#/definitions/Book",
+            "readOnly": true,
+        },
+        "create_time": {
+            "type": "string",
+            "format": "date-time",
+            "readOnly": true,
+        },
+        "aliases": {
+            "type": "array",
+            "items": {
+                "type": "string"
+            },
+            "readOnly": true,
+        }
+    },
+    "required": ["path"],
+}
+```
+
+{% endtabs %}
+
+- The resource revision **must** contain a field with a message type of the
+  parent resource, with a field name of `resource`.
+    - The value of `resource` **must** be the original resource
+      at the point in time the revision was created.
+- The resource revision **must** contain a `create_time` field (see [AIP-142][]).
+- The resource revision **may** contain a repeated field `aliases`, which would
+  contain a list of resource IDs that the revision is also known by (e.g. `latest`)
+
+### Creating Revisions
+
+Depending on the resource, different APIs may have different strategies for
+creating a new revision. Specifically examples of strategies include:
+
+- Creating a revision when there is a change to the resource
+- Creating a revision when important system state changes
+- Creating a revision when specifically requested
+
+APIs **may** use any of these strategies or any other strategy. APIs **must**
+document their revision creation strategy.
+
+APIs **should** have at least one revision for their resource at any time.
+
+### Resource names for revisions
+
+When referring to specific revision of a resource, the subcollection name
```

Comment:
```suggestion
When referring to the revisions of a resource, the subcollection name
```

Reply:
Do we also want something here about how the resource name of the revision resource must be "{Resource}Revision", i.e. "BookRevision"?

Otherwise this could also be interpreted as saying that the resource has to be named "Revision".  This obviously doesn't work for APIs with multiple revisioned resources...but doesn't hurt to spell it out.

Reply:
Are there exceptional cases to this? I guess not if the path includes `revision` in the name. Added.